ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java

```diff
@@ -58,40 +67,54 @@ public class EthPeers {
       Comparator.comparing(EthPeer::outstandingRequests)
           .thenComparing(EthPeer::getLastRequestTimestamp);
 
-  private final Map<PeerConnection, EthPeer> connections = new ConcurrentHashMap<>();
+  private final Map<Bytes, EthPeer> connections = new ConcurrentHashMap<>();
+
+  private final Cache<PeerConnection, EthPeer> incompleteConnections =
+      CacheBuilder.newBuilder()
+          .expireAfterWrite(Duration.ofSeconds(20L))
+          .concurrencyLevel(1)
+          .removalListener(this::onCacheRemoval)
+          .build();
   private final String protocolName;
   private final Clock clock;
   private final List<NodeMessagePermissioningProvider> permissioningProviders;
-  private final int maxPeers;
   private final int maxMessageSize;
   private final Subscribers<ConnectCallback> connectCallbacks = Subscribers.create();
   private final Subscribers<DisconnectCallback> disconnectCallbacks = Subscribers.create();
   private final Collection<PendingPeerRequest> pendingRequests = new CopyOnWriteArrayList<>();
+  private final int peerLowerBound;
+  private final int peerUpperBound;
+  private final int maxRemotelyInitiatedConnections;
+  private final Boolean randomPeerPriority;
+  private final Bytes nodeIdMask = Bytes.random(64);
```

Comment:
in the original this was a constant - message length? can we avoid hard coding it

Reply:
Done